logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/scenario/OnIncomingCall.kt

```diff
@@ -39,7 +39,8 @@ class OnIncomingCall(
                 status = CallStatus.INCOMING,
                 callerId = federatedIdMapper.parseToFederatedId(userId),
                 isMuted = isMuted,
-                isCameraOn = isVideoCall
+                isCameraOn = isVideoCall,
+                shouldRing = shouldRing
```

Comment:
Do we have to pass `shouldRing` to `createCall` ?
 we could check and pass the right status here before calling `createCall()`

Reply:
Yes, because the logic to handle the rest of the status is being done inside the `createCall` in `callRepository`, doing the logic to send the correct status already would have to refactor the whole `createCall` function and it will take some time and its not something we will want to do right now as we have more calling features to implement before fully diving deep on optimization on this code.